fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/sync/SyncType.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.attachment.sync;
+
+public enum SyncType {
```

Comment:
I dont really understand what this is for? Is this not dictated by the predicate?

Reply:
I suppose I could replace it with an identity check? It felt more convenient to check an enum to distinguish the cases.